extensions/markdown-language-features/src/util/topmostLineMonitor.ts

```diff
@@ -22,11 +35,50 @@ export class TopmostLineMonitor extends Disposable {
 				}
 			}
 		}));
+
+		this._register(vscode.window.onDidChangeActiveTextEditor(textEditor => {
+
+			// When at a markdown file, apply existing scroll settings from static preview if last editor was custom.
+			// Also save reference to text editor for line number reference later
+			if (textEditor && textEditor.document && isMarkdownFile(textEditor.document)) {
+				if (this.isPrevEditorCustom) {
+					const line = this.getPreviousStaticEditorLineByUri(textEditor.document.uri);
+					if (line) {
+						this._onEditorNeedsScrolling.fire({ line: line, editor: textEditor });
+					}
+				}
+				this.setPreviousMDTextEditorLine(textEditor);
+			}
+
+			this.isPrevEditorCustom = (textEditor === undefined);
+		}));
 	}
 
 	private readonly _onChanged = this._register(new vscode.EventEmitter<{ readonly resource: vscode.Uri, readonly line: number }>());
 	public readonly onDidChanged = this._onChanged.event;
 
+	private readonly _onEditorNeedsScrolling = this._register(new vscode.EventEmitter<{ readonly line: number, readonly editor: vscode.TextEditor }>());
+	public readonly onEditorNeedsScrolling = this._onEditorNeedsScrolling.event;
+
+	public setPreviousMDTextEditorLine(editor: vscode.TextEditor) {
+		const uri = editor.document.uri;
+		this.previousMDTextEditors.set(uri.toString(), editor);
```

Comment:
The `TextEditor` object becomes invalid as soon as that editor is closed, so it's kind of dangerous to keep a reference to it around

Instead you can use `window.visibleTextEditors` to get all current editors

Reply:
I think we need to keep information about closed text editors to get the scroll info when we switch editors. Since keeping text editors won't work, we can just keep the last scroll location as with the static preview scroll locations.